inc/add_site_ui/namespace.php

```diff
@@ -186,9 +188,46 @@ function output_add_site_page() {
  * @return bool True if valid, false otherwise.
  */
 function validate_domain_segment( string $segment ) : bool {
+	$segment = sanitize_domain_segment( $segment );
 	return (bool) preg_match( '/^' . REGEX_DOMAIN_SEGMENT . '$/', $segment );
 }
 
+/**
+ * Sanitize a domain segment.
+ *
+ * Strips out invalid characters from a URL segment.
+ *
+ * @param string $segment Dirty segment to sanitize.
+ *
+ * @return string The cleaned segment.
+ */
+function sanitize_domain_segment( string $segment ) : string {
+	$segment = wp_kses_no_null( $segment );
+	$segment = remove_accents( $segment );
```

Comment:
Given this is passed through `idna_encode()` later why remove accents here? Was there some reason not to use `idna_encode()` at this stage?

Reply:
That's a good point and might fix the `röb` turning into `rob` when entered. Left it in because it was in the HappyTables stuff, but our need is probably more advanced. If we're removing that, this function seems to serve no purpose anymore, so will probably pull it out and maybe put the `wp_kses` call elsewhere.